File: apps/web/lib/evaluation.ts
import { apiRequest } from "@/lib/api";

export type EvaluationRun = {
  id: string;
  name: string;
  status: string;
  suite_names: string[];
  dataset_version: string;
  dataset_checksum: string;
  model_metadata: Record<string, unknown>;
  prompt_version: string;
  retrieval_configuration: Record<string, unknown>;
  environment_metadata: Record<string, unknown>;
  aggregate_metrics: Record<string, unknown> | null;
  failure: Record<string, unknown> | null;
  created_at: string;
  started_at: string | null;
  completed_at: string | null;
};

export async function fetchEvaluationRuns(): Promise<EvaluationRun[]> {
  return apiRequest(
    "/api/v1/evaluation/runs",
    undefined,
    "Evaluation load failed",
  );
}

File: apps/api/app/schemas/evaluation.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class EvaluationRunResponse(BaseModel):
    id: UUID
    name: str
    status: str
    suite_names: list[str]
    dataset_version: str
    dataset_checksum: str
    model_metadata: dict[str, object]
    prompt_version: str
    retrieval_configuration: dict[str, object]
    environment_metadata: dict[str, object]
    aggregate_metrics: dict[str, object] | None
    failure: dict[str, object] | None
    created_at: datetime
    started_at: datetime | None
    completed_at: datetime | None


class EvaluationResultResponse(BaseModel):
    id: UUID
    evaluation_run_id: UUID
    test_case_id: str
    question_category: str | None
    expected_answerability: str | None
    deterministic_metrics: dict[str, object]
    judge_metrics: dict[str, object] | None
    passed: bool
    failure_reasons: list[str]
    latency_ms: float | None
    created_at: datetime
